### PR TITLE
Improve system pod qos/priorities

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

**Comment by jakolehm** [comment]
We need to add new priorityclass for system-level addons.

**Comment by jnummelin** [comment]
> We need to add new priorityclass for system-level addons.

I think that can be handled as separate PR

File: lib/pharos/resources/calico/daemonset.yml.erb [hunk 1]
```diff
@@ -38,6 +38,7 @@ spec:
         - effect: NoExecute
           operator: Exists
       serviceAccountName: calico-node
+      priorityClassName: system-cluster-critical
```
Comment (jnummelin) [inline lib/pharos/resources/calico/daemonset.yml.erb]: Maybe even `system-node-critical`

File: lib/pharos/scripts/configure-etcd.sh [hunk 2]
```diff
@@ -21,6 +21,7 @@ metadata:
   name: etcd
   namespace: kube-system
 spec:
+  priorityClassName: system-cluster-critical
```
Comment (jnummelin) [inline lib/pharos/scripts/configure-etcd.sh]: I think this needs to be `system-node-critical` that has the highest priority. I mean if etcd would get preempted, everything breaks. :) Although this is a static pod, so I don't think these will actually ever get preempted.

File: lib/pharos/resources/weave/daemon-set.yml.erb [hunk 3]
```diff
@@ -88,6 +88,7 @@ spec:
       securityContext:
         seLinuxOptions: {}
       serviceAccountName: weave-net
+      priorityClassName: system-cluster-critical
```
Comment (jnummelin) [inline lib/pharos/resources/weave/daemon-set.yml.erb]: Same as for Calico, I'd put these as `system-node-critical`

**Comment by jakolehm** [comment]
> I'd bump some of the priorities

Bumped.

**Pull request merged**